### ISSUE TITLE
Allow setting the level of every hint and warning

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/Nim#20905]
### Summary

For every warning and hint, allow setting the level to `None`, `Hint`, `Warning` or `Error` - basically, for every "optional" diagnostic, it should be possible to set its effect on the compilation process.

### Description

The objective is that codebases that have addressed all instances of a particular kind can set that particular message to `Error` and prevent new instances from creeping into the codebase - the difference between `Hint` and `Warning` in the compiler is somewhat subjective as well - some hints are appropriate as warnings/errors in some codebases while some warnings are not urgent in some contexts.

**Comment by Araq** [comment]
We already have `--warningAsError:warningX:on`, are you aware? :-)

**Comment by arnetheduck** [comment]
somewhat - but the way we interface with these is somewhat all over the place, specially when interacting with hints and style checks - ie the distinction between hint and warning is somewhat arbitrary, except for the "alert level" - `--styleCheck:error` for example sets the level of stylecheck "warnings" - the idea with this feature would be that there would be a single "notifications" mechanism, and for each message type in it, you could set the desired "level" - whatever flags set it would look like `--msg:MessageX:error --msg:[MessageY,MessageZ]:none` - then flags like `styleCheck` would have only `on/off/usages` - ie not serve the dual purpose as they do today.

Also, `--warningsAsError` doesn't allow me to turn warnings into hints or off - for that I have to use a different flag making the UX confusing (specially when taking into account features that were developed separately, like style checks).

**Comment by metagn** [comment]
What exactly is the classifying difference between warnings and hints? Is it that one is off by default while the other is on?

**Comment by Araq** [comment]
No, there is not much difference. A "hint" is benign, a "warning" deserves your attention.
